geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -231,107 +238,63 @@ object IngestCommand extends LazyLogging {
       inputs: Inputs,
       format: Option[String],
       ds: Option[_ <: DataStore]): Try[Option[(SimpleFeatureType, Config)]] = Try {
-    import org.locationtech.geomesa.utils.conversions.ScalaImplicits.RichIterator
-
     // try to load the sft, first check for an existing schema, then load from the params/environment
     var sft: SimpleFeatureType = loadSft(params, ds).orNull
 
     var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
 
     // if there is no converter passed in
-    if (converter == null && inputs.paths.nonEmpty) {
-      var open: () => InputStream = null
-      var file: FileHandle = null
-      var opened = ListBuffer.empty[CloseableIterator[InputStream]]
-
-      // try to infer the schema from stdin
-      if (inputs.paths == Inputs.StdInInputs) {
-        Command.user.info("No converter defined - will attempt to detect schema from stdin")
-
-        // Mark the beginning of the stdin stream so we can return to it later
-        System.in.mark(8192)
-
-        // Function that returns an input stream containing the cached bytes and any bytes remaining on stdin
-        def openStdin(): InputStream = {
-          try {
-            // Get the CachingStdInHandle that gives us access to our custom-made input stream
-            val handle = inputs.handles.headOption.getOrElse {
-              throw new RuntimeException("Unable to obtain handle for stdin")
-            }
-
-            // Use the handle to get the stream
-            val (_, stream) = handle.open.next()
-            opened += CloseableIterator.single(stream)
-            stream
-          } catch {
-            case e: Exception => throw new RuntimeException("Unable to read from stdin", e)
-          }
-        }
-
-        open = openStdin
+    if (converter == null && (inputs.stdin || inputs.paths.nonEmpty)) {
```

Comment:
Is this `if` statement refactoring mainly for the purpose of improving runtime by reducing the amount of branching in the code? Or is it just for readability?

Reply:
i don't think it's likely to improve runtime, but it avoids a lot of duplicate/very similar code between the 2 branches, and makes it easier to understand (at least IMO)